library/build.gradle

```diff
@@ -1,75 +1,66 @@
+apply plugin: 'maven-publish'
 
 archivesBaseName = 'marqueeview'
 group = 'asia.ivity.android'
 
 
+def  version = '1.0'
+
 android {
 	compileSdkVersion 30
 	buildToolsVersion "29.0.2"
-}
+	defaultConfig {
+		minSdkVersion 19
+		targetSdkVersion 30
+		versionCode 1
+		versionName(version)
+		testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+		}
 
-// TODO: Add a task that prints the test results.
-task instrumentTest(dependsOn: connectedCheck)
-
-signing {
-    sign configurations.archives
+	buildTypes {
+		release {
+			minifyEnabled false
+			proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+		}
+	}
 }
 
-if(hasProperty("sonatypeUsername") && hasProperty("sonatypePassword")) {
-    uploadArchives {
-        repositories {
-            mavenDeployer {
-                beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+publishing {
+	publications {
+		mavenJAR(MavenPublication) {
+			afterEvaluate {
+				artifact libraryJar
+			}
 
-                repository(url: 'https://oss.sonatype.org/service/local/staging/deploy/maven2') {
-                    authentication(userName: sonatypeUsername, password: sonatypePassword)
-                }
-                snapshotRepository(url: 'https://oss.sonatype.org/content/repositories/snapshots/') {
-                    authentication(userName: sonatypeUsername, password: sonatypePassword)
-                }
+			pom {
+				packaging = 'jar'
 
-                modifyPom(addFilter('aar') { artifact, file ->
-                    artifact.name == 'marqueeview'
-                })
-            }
-        }
-    }
-}
+				name = 'MarqueeView'
+				description = 'Extensive library for customized marquee effect.'
+				url = 'https://github.com/ened/Android-MarqueeView'
 
-def modifyPom(pom) {
-    pom.project {
-        name 'MarqueeView'
-        description 'Extensive library for customized marquee effect.'
-        url 'https://github.com/ened/Android-MarqueeView'
-
-        scm {
-            url 'https://github.com/ened/Android-MarqueeView.git'
-            connection 'git@github.com:ened/Android-MarqueeView.git'
-            developerConnection 'git@github.com:ened/Android-MarqueeView.git'
-        }
-
-        licenses {
-            license {
-                name 'BSD License'
-                url 'http://opensource.org/licenses/bsd-license.php'
-                distribution 'repo'
-            }
-        }
+				licenses {
+					license {
+						name = 'BSD License'
+						url = 'http://opensource.org/licenses/bsd-license.php'
+						distribution = 'repo'
+					}
+				}
+			}
+		}
+	}
+}
 
-        organization {
-            name 'Ivity Asia'
-            url 'http://www.ivity.asia'
-        }
+task libraryJar(type: Jar, dependsOn: ['compileReleaseKotlin']) {
+	from {
+		compileReleaseKotlin.destinationDir
+	}
+	android.sourceSets.all { sourceSet ->
+		from(sourceSet.assets.srcDirs) {
+			into "assets/"
+		}
+	}
```

Comment:
I don't think there are assets to include here (unlike KotliTeX), so this bit can probably be removed.

Reply:
Noted.